stripe/src/test/java/com/stripe/android/paymentsheet/PaymentSheetActivityTest.kt

```diff
@@ -85,11 +113,26 @@ class PaymentSheetActivityTest {
             assertThat(currentFragment(activity)).isInstanceOf(PaymentSheetPaymentMethodsListFragment::class.java)
 
             activity.onBackPressed()
+            idleLooper()
             // animating out
             assertThat(activity.bottomSheetBehavior.state).isEqualTo(BottomSheetBehavior.STATE_HIDDEN)
         }
     }
 
+    @Test
+    fun `handles buy button clicks`() {
+        val scenario = activityScenario()
+        scenario.launch(intent).onActivity { activity ->
+            // wait for bottom sheet to animate in
+            testCoroutineDispatcher.advanceTimeBy(500)
+            idleLooper()
+
+            viewModel.updateSelection(PaymentSelection.Saved("saved_pm"))
+            activity.viewBinding.buyButton.performClick()
+            verify(viewModel).checkout(activity)
```

Comment:
was hoping to be able to test this end-to-end but launching `PaymentRelayActivity` never seems to happen (robolectric seems to treat this as a no-op)

Reply:
Does `peekNextStartedActivity` work?

Reply:
idk why this wasn't working for me when I originally tried it 🤦